discovery_engine/test/integration/init_engine_test.dart

```diff
@@ -48,7 +48,7 @@ void main() {
 
     test('news api request error should not raise an engine init exception',
         () async {
-      server.replyWithError = true;
+      server.replyWith = ReplyWith.error;
```

Comment:
❓ was there a reason setting it this way is preferred?

Reply:
due to the newly introduced `server.replyWithEmpty` we would need to manage to exclusive booleans instead (only at  most one of `replyWithError` and `replyWithEmpty` must be true at the same time). hence i replaced it with an enum.

Reply:
I see, thanks